### PR TITLE
feat: define router -> ingester write protocol

### PR DESCRIPTION
**Pull request by domodwyer**
This PR defines a simple request/response placeholder for a direct RPC write from a router, to an ingester.

---

* feat: define router -> ingester write protocol (dcc7b10bc)

      Specify a gRPC service and request/response message formats to push
      writes directly from a router to an ingester.

### CODE REVIEW

File: generated_types/protos/influxdata/iox/ingester/v1/write.proto [hunk 1]
```diff
@@ -0,0 +1,15 @@
+syntax = "proto3";
+package influxdata.iox.ingester.v1;
+option go_package = "github.com/influxdata/iox/ingester/v1";
```
Comment (domodwyer) [inline generated_types/protos/influxdata/iox/ingester/v1/write.proto:3]: This defines the proto package + import path.

File: generated_types/protos/influxdata/iox/ingester/v1/write.proto [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+import "influxdata/pbdata/v1/influxdb_pb_data_protocol.proto";
```
Comment (domodwyer) [inline generated_types/protos/influxdata/iox/ingester/v1/write.proto:5]: This imports the existing write protocol data type (`DatabaseBatch`)

File: generated_types/build.rs [hunk 3]
```diff
@@ -54,6 +54,7 @@ fn generate_grpc_types(root: &Path) -> Result<()> {
         ingester_path.join("parquet_metadata.proto"),
         ingester_path.join("query.proto"),
         ingester_path.join("write_info.proto"),
+        ingester_path.join("write.proto"),
```
Comment (domodwyer) [inline generated_types/build.rs:57]: And this configures the rust build step to generate rust code from the proto file.

File: test_helpers_end_to_end/src/client.rs [hunk 4]
```diff
@@ -51,18 +50,6 @@ pub fn get_write_token(response: &Response<Body>) -> String {
         .to_string()
 }
 
-/// Extracts the write token from the specified response (to the gRPC write API)
-pub fn get_write_token_from_grpc(response: &tonic::Response<WriteResponse>) -> String {
```
Comment (domodwyer) [inline test_helpers_end_to_end/src/client.rs:55]: This is just dead code conflicting with names / leftover from removal of the old external gRPC write API in the router.

**Review by pauldix — APPROVED** [review]

**Comment by pauldix** [comment]
@aierui yes, we're removing Kafka from our setup. We've found that we don't need all the extra guarantees and functionality it provides. It also adds quite a bit of cost, so we're optimizing for a more efficient and dynamically scalable ingestion pipeline. I'll probably write a more public blog post about it once we've tested this at scale in our mirrored production environment.

**Pull request merged**